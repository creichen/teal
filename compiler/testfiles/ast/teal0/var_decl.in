// uninitilized variable declaration
var a  : any ;
/*
	initialized variable declarations
 */
var b : int := 0x40 ;

var c : int := 10 ;

var d : string := "escaped\" string" ;
