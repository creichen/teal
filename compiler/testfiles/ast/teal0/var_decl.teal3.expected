Module
  List
  List
    VarDecl
      IdDecl Identifier="a"
      Opt
        Type
          IdUse Identifier="any"
          List
          List
      Opt
    VarDecl
      IdDecl Identifier="b"
      Opt
        Type
          IdUse Identifier="int"
          List
          List
      Opt
        IntConstant Value="64"
    VarDecl
      IdDecl Identifier="c"
      Opt
        Type
          IdUse Identifier="int"
          List
          List
      Opt
        IntConstant Value="10"
    VarDecl
      IdDecl Identifier="d"
      Opt
        Type
          IdUse Identifier="string"
          List
          List
      Opt
        StringConstant Value="escaped\" string"
